crates/shadowsocks-service/src/local/tun/tcp.rs

```diff
@@ -469,24 +454,22 @@ impl TcpTun {
                                 .poll_delay(before_poll, &socket_set)
                                 .unwrap_or(SmolDuration::from_millis(5));
                             if next_duration != SmolDuration::ZERO {
-                                thread::park_timeout(Duration::from(next_duration));
+                                tokio::time::sleep(Duration::from(next_duration)).await;
```

Comment:
It may not be a good choice to make this loop running as a Task running in tokio's runtime, because it will use most of the CPU time of worker threads in the runtime, which will eventually starve the other tasks.

And this should not be changed to `sleep`. I use `park_timeout` here because we can use `unpark()` to wake up the thread and continue the loop. If it changed to `sleep`, then `poll_delay` will not be called immediately after sockets' status changed.

Reply:
After reviewing it, just modify it directly on my fork.

Reply:
Your commits are in this PR, if I haven't missed any other changes. This comment is based on your latest commit.